pkg/virt-operator/resource/apply/apps_test.go

```diff
@@ -1166,4 +1168,106 @@ var _ = Describe("Apply Apps", func() {
 		)
 	})
+
+	Context("on calling syncDeployment", func() {
+		var cachedDeployment *appsv1.Deployment
+		var strategyDeployment *appsv1.Deployment
+		var err error
+		var clientset *kubecli.MockKubevirtClient
+		var dpClient *fake.Clientset
+		var kv *v1.KubeVirt
+		var expectations *util.Expectations
+		var stores util.Stores
+		var ctrl *gomock.Controller
+		var revisionAnnotation = "deployment.kubernetes.io/revision"
+		var fakeAnnotation = "fakeAnnotation.io/fake"
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			kvInterface := kubecli.NewMockKubeVirtInterface(ctrl)
+			expectations = &util.Expectations{}
+			clientset = kubecli.NewMockKubevirtClient(ctrl)
+			dpClient = fake.NewSimpleClientset()
+			clientset.EXPECT().KubeVirt(Namespace).Return(kvInterface).AnyTimes()
+			clientset.EXPECT().AppsV1().Return(dpClient.AppsV1()).AnyTimes()
+
+			kv = &v1.KubeVirt{}
+			virtControllerConfig := &util.KubeVirtDeploymentConfig{
+				Registry:        Registry,
+				KubeVirtVersion: Version,
+			}
+			strategyDeployment, err = components.NewControllerDeployment(
+				Namespace,
+				virtControllerConfig.GetImageRegistry(),
+				virtControllerConfig.GetImagePrefix(),
+				virtControllerConfig.GetControllerVersion(),
+				virtControllerConfig.GetLauncherVersion(),
+				virtControllerConfig.GetExportServerVersion(),
+				"",
+				"",
+				"",
+				"",
+				virtControllerConfig.VirtControllerImage,
+				virtControllerConfig.VirtLauncherImage,
+				virtControllerConfig.VirtExportServerImage,
+				virtControllerConfig.SidecarShimImage,
+				virtControllerConfig.GetImagePullPolicy(),
+				virtControllerConfig.GetImagePullSecrets(),
+				virtControllerConfig.GetVerbosity(),
+				virtControllerConfig.GetExtraEnv())
+			Expect(err).ToNot(HaveOccurred())
+
+			cachedDeployment = strategyDeployment.DeepCopy()
+			cachedDeployment.Generation = 2
+			cachedDeployment.Annotations = map[string]string{
+				revisionAnnotation: "4",
+				fakeAnnotation:     "fake",
+			}
+
+			stores = util.Stores{DeploymentCache: &MockStore{get: cachedDeployment}}
+		})
+
+		It("should not remove revision annotation", func() {
+			dpClient.Fake.PrependReactor("patch", "deployments", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				cachedDeploymentBytes, err := json.Marshal(cachedDeployment)
+				if err != nil {
+					return false, nil, err
+				}
+
+				patchAction := action.(testing.PatchAction)
+				patchBytes := patchAction.GetPatch()
+
+				patchJSON, err := jsonpatch.DecodePatch(patchBytes)
+				if err != nil {
+					return false, nil, err
+				}
+
+				newDeploymentBytes, err := patchJSON.Apply(cachedDeploymentBytes)
+				if err != nil {
+					return false, nil, err
+				}
+
+				var newDeployment *appsv1.Deployment
+				if err = json.Unmarshal(newDeploymentBytes, &newDeployment); err != nil {
+					return false, nil, err
+				}
+
+				return true, newDeployment, nil
+			})
```

Comment:
You can avoid customizing the behavior and use the fake client directly (it already does this job for you).
It is enough to register the deployment in the fake client (through `Create`) and then check the resource after the `syncDeployment` function.
This is just an example for inspiration: https://github.com/fossedihelm/kubevirt/commit/b5632fd751e1833e76b44104e0a801b0917a4016

N.B: There are couple of switch betweem var/const and drop of unneeded global variables

Reply:
Thanks for the note i modified the test please have another loop :)